### PR TITLE
 Full French UI Localization

### PR DESCRIPTION
**Pull request by Korben00**
Hi friends, 

I've translated the Openpilot UI into French to improve accessibility for French-speaking users. The translation covers all UI elements to ensure a fully localized user experience.

### Verification
I've thoroughly tested the translated software by running it in the test scripts and cross-checking all translated texts with their English counterparts. No issues with character encoding or text overflow in the UI were found. Furthermore, I'm a native French-speaking and regular user of openpilot, so I can confirm the accuracy and appropriateness of the translations.

### Additional remarks
During my translation and testing, I noticed that some interface elements were not localized, such as the guide, certain messages in the "software" section like "checking," "downloading," "finalizing update," as well as all messages that appear during driving like "Experimental Branch," "Reverse Gear," etc.

K.

### CODE REVIEW

**Comment by Korben00** [comment]
Hi, 
I'd like to indicate in this PR that I'll be following future interface developments and proposing updates to this language file on an ongoing basis. With most of the work done, this will be easy.

**Comment by nworb-cire** [comment]
LGTM. Why « enlever » in one place but « désinstaller » elsewhere? Also, does « détente » carry a precisely identical connotation to "chill"? (Perhaps just a sign of my youth, but my first inclination would have been to anglicise this. Not saying it has to change.)

File: selfdrive/ui/translations/main_fr.ts [hunk 1]
```diff
@@ -0,0 +1,1158 @@
...
+    <message>
+        <source>Cancel</source>
+        <translation>Annuler</translation>
+    </message>
+</context>
+<context>
+    <name>DeclinePage</name>
+    <message>
+        <source>You must accept the Terms and Conditions in order to use openpilot.</source>
+        <translation>Vous devez accepter les conditions générales pour utiliser OpenPilot.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:108]:
```suggestion
        <translation>Vous devez accepter les conditions générales pour utiliser openpilot.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 2]
```diff
@@ -0,0 +1,1158 @@
...
+    <message>
+        <source>Decline, uninstall %1</source>
+        <translation>Refuser, désinstaller %1</translation>
+    </message>
+</context>
+<context>
+    <name>DevicePanel</name>
+    <message>
+        <source>Dongle ID</source>
+        <translation>ID du Dongle</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:123]:
```suggestion
        <translation>Dongle ID</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 3]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Enable Lane Departure Warnings</source>
+        <translation>Activer les avertissements de sortie de voie</translation>
+    </message>
+    <message>
+        <source>Receive alerts to steer back into the lane when your vehicle drifts over a detected lane line without a turn signal activated while driving over 31 mph (50 km/h).</source>
+        <translation>Recevez des alertes pour revenir dans la voie lorsque votre véhicule dérive au-delà d&apos;une ligne de voie détectée sans clignotant activé en roulant à plus de 31 mph (50 km/h).</translation>
+    </message>
+    <message>
+        <source>Record and Upload Driver Camera</source>
+        <translation>Enregistrer et uploader la vidéo conducteur</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:990]:
```suggestion
        <translation>Enregistrer et publier la vidéo conducteur</translation>
```

Comment (maxime-desroches) [reply selfdrive/ui/translations/main_fr.ts:990]: téléverser est un autre choix

File: selfdrive/ui/translations/main_fr.ts [hunk 4]
```diff
@@ -0,0 +1,1158 @@
...
+    </message>
+    <message>
+        <source>Upload data from the driver facing camera and help improve the driver monitoring algorithm.</source>
+        <translation>Téléchargez les données de la caméra orientée vers le conducteur et aidez à améliorer l&apos;algorithme de surveillance du conducteur.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:994]:
```suggestion
        <translation>Publiez les données de la caméra orientée vers le conducteur et aidez à améliorer l&apos;algorithme de surveillance du conducteur.</translation>
```

Comment (maxime-desroches) [reply selfdrive/ui/translations/main_fr.ts:994]: même chose

File: selfdrive/ui/translations/main_fr.ts [hunk 5]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Use 24h format instead of am/pm</source>
+        <translation>Utiliser le format 24h plutôt que am/pm</translation>
+    </message>
+    <message>
+        <source>Show Map on Left Side of UI</source>
+        <translation>Afficher la carte à gauche de l&apos;interface</translation>
+    </message>
+    <message>
+        <source>Show map on left side when in split screen view.</source>
+        <translation>Afficher la carte à gauche en mode écran partagé.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:1018]:
```suggestion
        <translation>Afficher la carte à gauche en mode écran scindé.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 6]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Connect to Wi-Fi to upload driving data and help improve openpilot</source>
+        <translation>Connectez-vous au Wi-Fi pour télécharger les données de conduite et aider à améliorer openpilot</translation>
+    </message>
+    <message>
+        <source>Open Settings</source>
+        <translation>Ouvrir les paramètres</translation>
+    </message>
+    <message>
+        <source>Uploading training data</source>
+        <translation>Envoi des données</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:1128]:
```suggestion
        <translation>Publication des données</translation>
```

Comment (maxime-desroches) [reply selfdrive/ui/translations/main_fr.ts:1128]: "Téléversement des données" est un autre choix

File: selfdrive/ui/translations/main_fr.ts [hunk 7]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>SELECT</source>
+        <translation>CHOISIR</translation>
+    </message>
+    <message>
+        <source>Select a branch</source>
+        <translation>Choisir une branche</translation>
+    </message>
+    <message>
+        <source>Uninstall %1</source>
+        <translation>Enlever %1</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:846]:
```suggestion
        <translation>Désinstaller %1</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 8]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Enable Roaming</source>
+        <translation>Activer l&apos;itinérance</translation>
+    </message>
+    <message>
+        <source>APN Setting</source>
+        <translation>Paramètre APN</translation>
+    </message>
+    <message>
+        <source>Enter APN</source>
+        <translation>Entrez l&apos;APN</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:55]:
```suggestion
        <translation>Entrer le nom du point d&apos;accès</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 9]
```diff
@@ -0,0 +1,1158 @@
...
+    <message>
+        <source>Disengage to Power Off</source>
+        <translation>Désengager pour éteindre</translation>
+    </message>
+</context>
+<context>
+    <name>DriveStats</name>
+    <message>
+        <source>Drives</source>
+        <translation>Conduites</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:254]:
```suggestion
        <translation>Trajets</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 10]
```diff
@@ -0,0 +1,1158 @@
...
+    </message>
+    <message>
+        <source>Hours</source>
+        <translation>Heures</translation>
+    </message>
+    <message>
+        <source>ALL TIME</source>
+        <translation>TOUJOURS</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:262]:
```suggestion
        <translation>DEPUIS TOUJOURS</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 11]
```diff
@@ -0,0 +1,1158 @@
...
+    </message>
+    <message>
+        <source>PAST WEEK</source>
+        <translation>SEMAINE DERNIÈRE</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:266]:
```suggestion
        <translation>CETTE SEMAINE</translation>
```

File: selfdrive/ui/translations/main_fr.ts (hunk 9 above)
Comment (maxime-desroches) [reply selfdrive/ui/translations/main_fr.ts:254]: Difficile à traduire exactement en un mot.
 
@nworb-cire suggestion?

File: selfdrive/ui/translations/main_fr.ts [hunk 12]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Pair your device to your comma account</source>
+        <translation>Associez votre appareil à votre compte comma</translation>
+    </message>
+    <message>
+        <source>Go to https://connect.comma.ai on your phone</source>
+        <translation>Allez sur https://connect.comma.ai sur votre téléphone</translation>
+    </message>
+    <message>
+        <source>Click &quot;add new device&quot; and scan the QR code on the right</source>
+        <translation>Cliquez sur &quot;ajouter un nouvel appareil&quot; et scannez le QR code à droite</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:465]:
```suggestion
        <translation>Cliquez sur &quot;ajouter un nouvel appareil&quot; et scannez le code QR à droite</translation>
```

File: selfdrive/ui/translations/main_fr.ts (hunk 9 above)
Comment (nworb-cire) [reply selfdrive/ui/translations/main_fr.ts:254]: Oui, je dirais que trajets marche mieux aussi

File: selfdrive/ui/translations/main_fr.ts [hunk 13]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Reboot</source>
+        <translation>Redémarrer</translation>
+    </message>
+    <message>
+        <source>Exit</source>
+        <translation>Quitter</translation>
+    </message>
+    <message>
+        <source>dashcam</source>
+        <translation>caméra</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:533]:
```suggestion
        <translation>dashcam</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 14]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Reboot</source>
+        <translation>Redémarrer</translation>
+    </message>
+    <message>
+        <source>Confirm</source>
+        <translation>Confirmer</translation>
+    </message>
+    <message>
+        <source>Unable to mount data partition. Partition may be corrupted. Press confirm to erase and reset your device.</source>
+        <translation>Impossible de monter la partition de données. La partition peut être corrompue. Appuyez sur confirmer pour effacer et réinitialiser votre appareil.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:599]:
```suggestion
        <translation>Impossible de monter la partition data. La partition peut être corrompue. Appuyez sur confirmer pour effacer et réinitialiser votre appareil.</translation>
```

Comment (maxime-desroches) [reply selfdrive/ui/translations/main_fr.ts:599]: "data" est le nom de la partition

File: selfdrive/ui/translations/main_fr.ts [hunk 15]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>No custom software found at this URL.</source>
+        <translation>Aucun logiciel personnalisé trouvé à cette URL.</translation>
+    </message>
+    <message>
+        <source>WARNING: Low Voltage</source>
+        <translation>ATTENTION : Tension faible</translation>
+    </message>
+    <message>
+        <source>Power your device in a car with a harness or proceed at your own risk.</source>
+        <translation>Alimentez votre appareil dans une voiture avec un faisceau ou continuez à vos risques et périls.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:649]:
```suggestion
        <translation>Alimentez votre appareil dans une voiture avec un harness ou continuez à vos risques et périls.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 16]
```diff
@@ -0,0 +1,1158 @@
...
+</context>
+<context>
+    <name>Sidebar</name>
+    <message>
+        <source>CONNECT</source>
+        <translation>RÉSEAU</translation>
+    </message>
+    <message>
+        <source>OFFLINE</source>
+        <translation>INDISPO</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:731]:
```suggestion
        <translation>HORS LIGNE</translation>
```

File: selfdrive/ui/translations/main_fr.ts (hunk 9 above)
Comment (nworb-cire) [reply selfdrive/ui/translations/main_fr.ts:254]: J'ai pas vu si https://github.com/commaai/openpilot/pull/28476 a enlevé les statistiques pour tout le monde, ou seulement pour les abonnés-- si pour tous, pas besoin de s'en soucier

File: selfdrive/ui/translations/main_fr.ts [hunk 17]
```diff
@@ -0,0 +1,1158 @@
...
+    </message>
+    <message>
+        <source>Try the Navigation Beta</source>
+        <translation>Essayez la navigation beta</translation>
+    </message>
+    <message>
+        <source>Get turn-by-turn directions displayed and more with a comma
+prime subscription. Sign up now: https://connect.comma.ai</source>
+        <translation>Obtenez des instructions de navigation et plus encore avec un abonnement comma
+prime. Inscrivez-vous maintenant : https://connect.comma.ai</translation>
```
Comment (nworb-cire) [inline selfdrive/ui/translations/main_fr.ts:380]: abonnez-vous est un autre choix

File: selfdrive/ui/translations/main_fr.ts [hunk 18]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Terms &amp; Conditions</source>
+        <translation>Termes &amp; Conditions</translation>
+    </message>
+    <message>
+        <source>Decline</source>
+        <translation>Refuser</translation>
+    </message>
+    <message>
+        <source>Scroll to accept</source>
+        <translation>Défiler pour accepter</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:939]:
```suggestion
        <translation>Faire défiler pour accepter</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 19]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Experimental Mode</source>
+        <translation>Mode expérimental</translation>
+    </message>
+    <message>
+        <source>Disengage on Accelerator Pedal</source>
+        <translation>Désengager avec la pédale d&apos;accélérateur</translation>
+    </message>
+    <message>
+        <source>When enabled, pressing the accelerator pedal will disengage openpilot.</source>
+        <translation>Lorsqu&apos;il est activé, appuyer sur la pédale d&apos;accélérateur désengage openpilot.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:978]:
```suggestion
        <translation>Lorsqu&apos;il est activé, appuyer sur la pédale d&apos;accélérateur désengagera openpilot.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 20]
```diff
@@ -0,0 +1,1158 @@
...
+        <translation>Laisse le modèle de conduite contrôler l&apos;accélérateur et les freins. openpilot conduira comme il pense qu&apos;un humain le ferait, y compris s&apos;arrêter aux feux rouges et aux panneaux stop. Comme le modèle de conduite décide de la vitesse à adopter, la vitesse définie ne servira que de limite supérieure. Cette fonctionnalité est en version alpha, des erreurs sont à prévoir.</translation>
+    </message>
+    <message>
+        <source>New Driving Visualization</source>
+        <translation>Nouvelle visualisation de la conduite</translation>
+    </message>
+    <message>
+        <source>The driving visualization will transition to the road-facing wide-angle camera at low speeds to better show some turns. The Experimental mode logo will also be shown in the top right corner.</source>
+        <translation>La visualisation de la conduite passera à la caméra grand angle orientée vers la route à faible vitesse pour mieux montrer certains virages. Le logo du mode expérimental sera également affiché dans le coin supérieur droit.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:1058]:
```suggestion
        <translation>À faible vitesse, la visualisation de la conduite utilisera la caméra à grand angle orientée vers la route pour mieux montrer certains virages. Le logo du mode expérimental sera également affiché dans le coin supérieur droit.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 21]
```diff
@@ -0,0 +1,1158 @@
...
+    </message>
+    <message>
+        <source>Experimental mode is currently unavailable on this car since the car&apos;s stock ACC is used for longitudinal control.</source>
+        <translation>Le mode expérimental est actuellement indisponible sur cette voiture car le régulateur de vitesse adaptatif d&apos;origine est utilisé pour le contrôle longitudinal.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:1062]:
```suggestion
        <translation>Le mode expérimental est actuellement indisponible pour cette voiture car le régulateur de vitesse adaptatif d&apos;origine est utilisé pour le contrôle longitudinal.</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 22]
```diff
@@ -0,0 +1,1158 @@
...
+</context>
+<context>
+    <name>WiFiPromptWidget</name>
+    <message>
+        <source>Setup Wi-Fi</source>
+        <translation>Activez le Wi-Fi</translation>
+    </message>
+    <message>
+        <source>Connect to Wi-Fi to upload driving data and help improve openpilot</source>
+        <translation>Connectez-vous au Wi-Fi pour télécharger les données de conduite et aider à améliorer openpilot</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:1120]:
```suggestion
        <translation>Connectez-vous au Wi-Fi pour publier les données de conduite et aider à améliorer openpilot</translation>
```

File: selfdrive/ui/translations/main_fr.ts [hunk 23]
```diff
@@ -0,0 +1,1158 @@
...
+        <source>Reboot</source>
+        <translation>Redémarrer</translation>
+    </message>
+    <message>
+        <source>Power Off</source>
+        <translation>Éteindre</translation>
+    </message>
+    <message>
+        <source>openpilot requires the device to be mounted within 4° left or right and within 5° up or 8° down. openpilot is continuously calibrating, resetting is rarely required.</source>
+        <translation>openpilot nécessite que l&apos;appareil soit installé à 4° à gauche ou à droite et à 5° vers le haut ou 8° vers le bas. openpilot se calibre en continu, la réinitialisation est rarement nécessaire.</translation>
```
Comment (maxime-desroches) [inline selfdrive/ui/translations/main_fr.ts:211]:
```suggestion
        <translation>openpilot nécessite que l&apos;appareil soit installé 4° à gauche ou à droite et à 5° vers le haut ou 8° vers le bas. openpilot se calibre en continu, la réinitialisation est rarement nécessaire.</translation>
```

**Comment by Korben00** [comment]
Thanks for your help with the translation. 👍

Have you been able to check that everything fits into the interface?

Thanks.

**Comment by Korben00** [comment]
Indeed, chill suits her too even if I'm not so young. 😎
As for the word "remove", it's mainly for reasons of space in the button, as uninstall was a bit too big.

**Comment by Korben00** [comment]
@sshane Hi! What are the next steps concerning this PR? Thank you.

**Comment by Korben00** [comment]
I'm sorry, I made a mistake on my side, on my fork, and it triggered a sync I wasn't expecting. I'm not expert enough with GitHub yet. In short, I apologise profusely for the miss click.

**Comment by sshane** [comment]
Looks like you deleted your fork. You can make it again by editing the file on this repo, then copy and pasting your changes. Let us know if you need any help

**Comment by Korben00** [comment]
Thanks @sshane . I followed your advice. 

New PR is here #29260

**Pull request closed without merging**